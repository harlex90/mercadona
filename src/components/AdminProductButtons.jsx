import { useIsAuthenticated } from "react-auth-kit";
import { useNavigate } from "react-router-dom";

const AdminProductButtons = () => {
    const navigate = useNavigate();
    const isAuthenticated = useIsAuthenticated();
    
    const onPlus = () => {
        navigate('/admin/products/create')
    }

    const onMinus = () => {
        navigate("/admin/products/delete")
    }

    const onPromotion = () => {
        navigate('/admin/promos/create')
    }

    if (!isAuthenticated()) return null;

    return(
        <div style={{display: "flex", flexDirection: "column", justifyContent: "center", alignItems: "center"}}>
            <div style={{display: "flex", justifyContent: "center", width: "100%", alignItems: "center", gap: "50px"}}>
                <button type='button' style={{display:"flex", justifyContent: "center", alignItems: "center", fontSize: "80px", backgroundColor: "white", width: "80px", height: "80px", borderRadius: "50%", border: "2px solid black"}}>
                    <a href="#" onClick={onPlus} style={{textDecoration: "none"}}>+</a>
                </button>
                <button type='button' style={{display:"flex", justifyContent: "center", alignItems: "center", fontSize: "80px", backgroundColor: "white", width: "80px", height: "80px", borderRadius: "50%", border: "2px solid black"}}>
                    <a href="#" onClick={onMinus} style={{textDecoration: "none"}}>-</a>
                </button>
                <button type='button' style={{display:"flex", justifyContent: "center", alignItems: "center", fontSize: "15px", backgroundColor: "white", width: "80px", height: "80px", borderRadius: "50%", border: "2px solid black"}}>
                    <a href="#" onClick={onPromotion} style={{textDecoration: "none"}}>Promotion</a>
                </button>
            </div>

        </div>

    )
}

export default AdminProductButtons;