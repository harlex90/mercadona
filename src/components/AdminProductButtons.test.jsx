import { render } from '@testing-library/react';
import AdminProductButtons from './AdminProductButtons';
import { useIsAuthenticated } from "react-auth-kit";

jest.mock('react-auth-kit', () => ({
    useIsAuthenticated: jest.fn(),
    useSignOut: () => () => console.log('signOut()')
}))

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    useNavigate: () => mockNavigate
}))

describe('AdminProductButtons', () => {
    test('match snapshot', () => {
        useIsAuthenticated.mockImplementation(() => () => false)
        const { asFragment } = render(<AdminProductButtons />);
        expect(asFragment()).toMatchSnapshot()
    });

    describe('authenticated', () => {
        beforeEach(() => {
            useIsAuthenticated.mockImplementation(() => () => true)
        })

        test('match snapshot', () => {
            const { asFragment } = render(<AdminProductButtons />);
            expect(asFragment()).toMatchSnapshot()
        });

        test('should navigate to /adm in/products/create if "+" button is clicked', () => {
           const { getByText } = render(<AdminProductButtons />);

            const buttonPlus = getByText('+');
            buttonPlus.click();
            expect(mockNavigate).toHaveBeenCalledWith("/admin/products/create")
        });
    
        test('should navigate to /admin/products/delete if "-" button is clicked', () => {            
            const { getByText } = render(<AdminProductButtons />);
            const buttonMinus = getByText('-');
            buttonMinus.click();
            expect(mockNavigate).toHaveBeenCalledWith("/admin/products/delete")
        });
    })


    
});
