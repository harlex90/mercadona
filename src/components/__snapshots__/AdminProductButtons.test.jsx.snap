// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminProductButtons authenticated match snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: flex; flex-direction: column; justify-content: center; align-items: center;"
  >
    <div
      style="display: flex; justify-content: center; width: 100%; align-items: center; gap: 50px;"
    >
      <button
        style="display: flex; justify-content: center; align-items: center; font-size: 80px; background-color: white; width: 80px; height: 80px; border-radius: 50%; border: 2px solid black;"
        type="button"
      >
        <a
          href="#"
          style="text-decoration: none;"
        >
          +
        </a>
      </button>
      <button
        style="display: flex; justify-content: center; align-items: center; font-size: 80px; background-color: white; width: 80px; height: 80px; border-radius: 50%; border: 2px solid black;"
        type="button"
      >
        <a
          href="#"
          style="text-decoration: none;"
        >
          -
        </a>
      </button>
      <button
        style="display: flex; justify-content: center; align-items: center; font-size: 15px; background-color: white; width: 80px; height: 80px; border-radius: 50%; border: 2px solid black;"
        type="button"
      >
        <a
          href="#"
          style="text-decoration: none;"
        >
          Promotion
        </a>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AdminProductButtons match snapshot 1`] = `<DocumentFragment />`;
